### ISSUE TITLE
[Bug]: Poor screenshot quality with PNG

### ISSUE THREAD

**Issue by ryno1234** [puppeteer/puppeteer#12721]
### Minimal, reproducible example

```TypeScript
const puppeteer = require('puppeteer');
const tmp = require('tmp');

let browser = null;
const contentUrl = "https://www.loremipsum.de/about_lorem_ipsum.html";

function start(){
    tmp.file(function _tempFileCreated(err, path, fd, cleanupCallback) {
        screenshotUrl(contentUrl, 1024, 768, path);
    });
}

async function screenshotUrl(url,  width, height, outputPath){
    const page = await browser.newPage();

    await page.setViewport({
        width,
        height
    });

    await page.goto(url, {"waitUntil" : "networkidle0"});

    await page.screenshot({
        path: outputPath,
        type: 'png',
        omitBackground: true,
        clip: {
            x: 0,
            y: 0,
            width,
            height
        }
    });

    console.log(`Wrote file out to ${outputPath}`);
    
    await page.close();
}


puppeteer.launch({args: ['--no-sandbox', '--disable-setuid-sandbox']})
    .then(b => {
        browser = b;
        start();
    });
```

### Error string

no error

### Bug behavior

- [ ] Flaky
- [ ] PDF

### Background

I'm looking to take a screenshot of a dynamically generated award.

### Expectation

The award to look good and represent the quality we're expecting

### Reality

The PNG output has significant image degradation. This is particularly seen in the text where it is easy to compare the original image HTML page (left) with the generated image (right).

The image below is in the actual use case, however it is not possible to provide these URLs publicly.

![image](https://github.com/puppeteer/puppeteer/assets/92051/3516c19a-d607-422b-b7ab-7e05b99cbac4)

### Puppeteer version

22.12.1

### Node version

18.14.1

### Package manager

npm

### Package manager version

9.3.1

### Operating system

Windows

**Comment by OrKoN** [comment]
Would you mind filing the issue at https://crbug.com/new as Puppeteer just saves screenshots done by the browser (via https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-captureScreenshot). Also, have you tried using jpeg with quality 100?

**Comment by AgCaliva** [comment]
Same issue as:
https://github.com/puppeteer/puppeteer/issues/1469
Puppeteer uses:
https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-captureScreenshot
I have observed that seems to work ok with full viewport screenshots(almost identical), but with specific element screenshots it looses some quality and image is a little more fuzzy, at least in Windows 10. Meaning that this problem its probably something in navigator implementation, puppeter should add some function to use your own screenshot software instead in my opinion, but that will probably require headless: false.
I have only tested using png.
I suggest to add a function to get element clip so we can manage to take the screenshot using an external software. Ty.
